server/www/classes/plotRelatedClasses/DataWidget.js

```diff
@@ -0,0 +1,104 @@
+var widgetCount = 0; // counts the number of DataWidgets instanciated, it is useful for its ids
+
+/*
+if the resize button of a certain widget is clicked 
+then, widgetBeingResized will contain a copy of this particular widget
```

Comment:
In JS, `var a = objectB;` is a reference to objectB, not a copy. (and this is very useful!)